### PR TITLE
[compiler-rt][www] replace deprecated LLVM_CONFIG_PATH with LLVM_CMAKE_DIR

### PR DESCRIPTION
**Pull request by c-rhodes**
This updates the standalone build docs for compiler-rt to replace deprecated LLVM_CONFIG_PATH with LLVM_CMAKE_DIR. A warning (added in D137024) is emitted for the current instructions.

### CODE REVIEW

**Review by llvm-beanz — APPROVED** [review]

File: compiler-rt/www/index.html [hunk 1]
```diff
@@ -123,7 +123,7 @@ <h2>Get it and get involved!</h2>
   <li>cd llvm-project</li>
   <li>mkdir build-compiler-rt</li>
   <li>cd build-compiler-rt</li>
-  <li>cmake ../compiler-rt -DLLVM_CONFIG_PATH=/path/to/llvm-config</li>
+  <li>cmake ../compiler-rt -DLLVM_CMAKE_DIR=/path/to/llvm-project/cmake</li>
```
Comment (llvm-beanz) [inline compiler-rt/www/index.html:126]:
```suggestion
  <li>cmake ../compiler-rt -DLLVM_CMAKE_DIR=/path/to/llvm-project/cmake/modules</li>
```

Comment (c-rhodes) [reply compiler-rt/www/index.html:126]: `llvm-project/cmake` worked for me when I built standalone but I see elsewhere modules is also specified

File: compiler-rt/www/index.html [hunk 2]
```diff
@@ -123,7 +123,7 @@ <h2>Get it and get involved!</h2>
   <li>cd llvm-project</li>
   <li>mkdir build-compiler-rt</li>
   <li>cd build-compiler-rt</li>
-  <li>cmake ../compiler-rt -DLLVM_CONFIG_PATH=/path/to/llvm-config</li>
+  <li>cmake ../compiler-rt -DLLVM_CMAKE_DIR=/path/to/llvm-project/cmake/modules</li>
```
Comment (vitalybuka) [inline compiler-rt/www/index.html:126]: It still mentions llvm-config on the line 120
and this instruction does not work for me.

Comment (vitalybuka) [reply compiler-rt/www/index.html:126]: It does not work `-DCOMPILER_RT_INCLUDE_TESTS=ON` (which I guess default):
-- Builtin supported architectures: 
CMake Error at test/CMakeLists.txt:47 (umbrella_lit_testsuite_begin):
  Unknown CMake command "umbrella_lit_testsuite_begin".

but work with `-DCOMPILER_RT_INCLUDE_TESTS=OFF`

Comment (vitalybuka) [reply compiler-rt/www/index.html:126]: No, it still does not work - `ninja` just copy a few headers and that's it

`-DLLVM_CMAKE_DIR=/full_path_to_llvm_build_dir/` works as expected, but I am not sure if that was expected.

```
cd llvm-project/
cmake -GNinja -B ../out/compiler-rt -S ./compiler-rt -DLLVM_CMAKE_DIR=$(readlink -f ../out/llvm_build) -DCMAKE_C_COMPILER=$(readlink -f ../out/llvm_build/bin/clang) -DCMAKE_CXX_COMPILER=$(readlink -f ../out/llvm_build/bin/clang++)
ninja -C ../out/compiler-rt
ninja -C ../out/compiler-rt check-all
```

Comment (c-rhodes) [reply compiler-rt/www/index.html:126]: > It does not work -DCOMPILER_RT_INCLUDE_TESTS=ON (which I guess default):
> -- Builtin supported architectures:
> CMake Error at test/CMakeLists.txt:47 (umbrella_lit_testsuite_begin):
> Unknown CMake command "umbrella_lit_testsuite_begin".
> 
> but work with -DCOMPILER_RT_INCLUDE_TESTS=OFF

I don't think `COMPILER_RT_INCLUDE_TESTS=ON` is always the default, only if `LLVM_TREE_AVAILABLE`:
https://github.com/llvm/llvm-project/blob/8fcb822da632ab4330b47641826ace01af0768e0/compiler-rt/cmake/base-config-ix.cmake#L35-L41

I can reproduce the error you're seeing when `COMPILER_RT_INCLUDE_TESTS=ON`:

> 
> cmake -G Ninja -S compiler-rt -B build-compiler-rt -DCMAKE_C_COMPILER=$HOME/clang+llvm-17.0.4-aarch64-linux-gnu/bin/clang-17 -DCMAKE_CXX_COMPILER=$HOME/clang+llvm-17.0.4-aarch64-linux-gnu/bin/clang++ -DLLVM_CMAKE_DIR=cmake -DCOMPILER_RT_INCLUDE_TESTS=ON
> -- Builtin supported architectures:
> Traceback (most recent call last):
>   File "<string>", line 22, in <module>
> IndexError: list index out of range
> CMake Error at test/CMakeLists.txt:47 (umbrella_lit_testsuite_begin):
>   Unknown CMake command "umbrella_lit_testsuite_begin".

but the fix proposed in #98707 doesn't fix it for me.

Comment (vitalybuka) [reply compiler-rt/www/index.html:126]: `-DLLVM_CMAKE_DIR=cmake` must be absolute path

Comment (c-rhodes) [reply compiler-rt/www/index.html:126]: > `-DLLVM_CMAKE_DIR=cmake` must be absolute path to the llvm/clang build dir so probably `-DLLVM_CMAKE_DIR=$HOME/clang+llvm-17.0.4-aarch64-linux-gnu` for you

that indeed did the trick, cheers

**Pull request merged**